### ISSUE TITLE
Logging uncaught errors to Microsoft.Extensions.Logging

### ISSUE THREAD

**Issue by ThomasArdal** [CoreWCF/CoreWCF#758]
The new [logging sample](https://github.com/CoreWCF/samples/tree/main/Scenarios/Logging-and-Dependency-Injection) is great. But after having a play with it, I realized that CoreWCF doesn't log uncaught Microsoft.Extensions.Logging (`ILogger`). Is there any plan to do so? In the meantime, we can implement an `IErrorHandler` and inject the `ILogger` in there to log uncaught exceptions. But I believe it would make sense to log errors to `ILogger` when they happen (like in ASP.NET Core).

**Comment by mconnew** [comment]
I understand where you're coming from. Like many things in CoreWCF, the solution to this isn't as simple as you would think. With something like WebMVC or WebAPI, the service implementation throwing an exception is a dead end whereas with CoreWCF, throwing an exception can be the actual response.
The way WCF deals with an unhandled exception in .NET Framework is to emit an event. If you have a trace source configured, this means it will be saved in an XML file. It will emit an event into the eventlog too. I need to work out an abstraction which will work with the ILogger way of doing things as well as keep doing things the "old" way. ILogger does have some abstractions built on top of it for structured logging where you know the field names of fields in the log. I need to investigate it further to see if that would be sufficiently performant as a way to bridge into logging to ETW for example.  

Basically once we add any logging built in to CoreWCF, we can't take it away without causing some people problems. Until we have the final solution worked out, adding any built-in logging will end up being legacy behavior we need to maintain going forwards and I want to avoid that ongoing overhead and cost.

**Comment by ThomasArdal** [comment]
Thank you for your elaborate response. Really appreciated 🙌 I'll close the task.
